hydrator-common/src/main/java/co/cask/hydrator/common/batch/JobUtils.java

```diff
@@ -28,14 +29,24 @@
  */
 public final class JobUtils {
 
+  private static final String MAPR_FS_IMPLEMENTATION_KEY = "fs.maprfs.impl";
+
   /**
    * Creates a new instance of {@link Job}. Note that the job created is not meant for actual MR
    * submission. It's just for setting up configurations.
    */
   public static Job createInstance() throws IOException {
     Job job = Job.getInstance();
     Configuration conf = job.getConfiguration();
+    // Remember the values of the default filesystem and implementation of the filesystem
+    // for MapR clusters. The values are used by FileInputFormat class while configuring the job.
+    String fsDefaultURI = conf.get(FileSystem.FS_DEFAULT_NAME_KEY);
+    String maprfsImplValue = conf.get(MAPR_FS_IMPLEMENTATION_KEY);
```

Comment:
For non mapr cluster, the value would be null, and we won't be setting it down below.

BTW @sagarkapare I think it's better to clear these two options (probably through the configurer) when the input provider is set.